python/ray/util/sgd/torch/worker_group.py

```diff
@@ -135,16 +137,47 @@ def __init__(self, max_workers, params, dist_params, initialization_hook,
         # The last time when this worker group was resized.
         self._last_resize = float("-inf")
 
+        # This is set only if a placement group is created by the worker group.
+        self._worker_placement_group = None
+
+    def _create_placement_group(self, num_workers):
+        """Creates a placement group for the workers.
+
+        If this worker is already in a placement group then a new one will
+        not be created. This is primarily for when Tune is the upstream and
+        will allocate resources for SGD workers.
+
+        If this worker is not in a placement group, a new one will be created
+        and set. The placement group will have a single bundle for each worker
+        and use the SPREAD strategy for an even distribution.
+        """
+        pg = get_current_placement_group()
+        if pg is None:
+            bundle = {
+                "CPU": self._num_cpus_per_worker,
+                "GPU": int(self._use_gpu)
+            }
+            bundles = [bundle] * num_workers
+            pg = ray.util.placement_group(bundles, strategy="SPREAD")
+            logger.debug("Waiting for placement group to start.")
+            ray.get(pg.ready(), timeout=self._timeout_s)
```

Comment:
This time out is used for torch process group creation:

```
 timeout_s (float): Seconds before the torch process group
            times out. Useful when machines are unreliable. If not set, default
            to 30 min, which is the same default as
            ``torch.init_process_group(...)``.
```

The default of 30 min doesn't really make sense for placement group creation. I think we could set this to 100 seconds by default and make it configurable via an environment variable (just like what XGBoost-Ray is doing).

You could add this to `sgd/torch/constants.py`. Something like 
```
SGD_PG_TIMEOUT = env_integer("SGD_PG_TIMEOUT", 100)
```

Reply:
Good idea! I added this but ran into some trouble unit testing, since the values in `constants.py` seem to be resolved before the test code is executed (so I can't set `os.environ["SGD_PLACEMENT_GROUP_TIMEOUT_S"]` to a smaller value). The test still runs successfully now, but takes 100 seconds. Any ideas on how to push this value through?

Reply:
Mocked the value with `patch` decorator, marking this as resolved now.